### PR TITLE
fix: Normalize keys of headers from server side requests

### LINKED ISSUE #1127: validation: throw exception if content-type is missing
**Describe the bug**
I get this error since upgrading to SK next-81 (also happens on 83). This possibly happens because of an API route with POST requests like so:

```ts
import fetch from 'node-fetch'


/**
 * Load data from API
 */
// Get results from a POST request
export async function post ({ headers, query, body, params, ...rest }) {
    try {
        const res = await fetch(`https://${import.meta.env.VITE_SANITY_ID}.apicdn.sanity.io/v1/data/query/production?query=${encodeURIComponent(body)}`, {
            method: 'GET',
            headers: {
                Authorization: `Bearer ${import.meta.env.VITE_SANITY_TOKEN}`,
            }
        })
        const data = await res.json()

        return {
            status: 200,
            body: data
        }
    }
     catch (error) {
        return {
            status: error.status || 500,
            body: error.message || error.text || `Can't fetch query`
        }
    }
}
```

**Logs**
```
Cannot read property 'split' of undefined
TypeError: Cannot read property 'split' of undefined
    at parse_body (file:///site/node_modules/.pnpm/@sveltejs+kit@1.0.0-next.83_svelte@3.37.0+vite@2.1.5/node_modules/@sveltejs/kit/dist/ssr.js:1330:60)
    at ssr (file:///site/node_modules/.pnpm/@sveltejs+kit@1.0.0-next.83_svelte@3.37.0+vite@2.1.5/node_modules/@sveltejs/kit/dist/ssr.js:1448:9)
    at fetch (file:///site/node_modules/.pnpm/@sveltejs+kit@1.0.0-next.83_svelte@3.37.0+vite@2.1.5/node_modules/@sveltejs/kit/dist/ssr.js:697:30)
```

**Information about your SvelteKit Installation:**

<details>
  <summary>Diagnostics</summary>

```
  System:
    OS: macOS 11.2.3
    CPU: (12) x64 Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz
    Memory: 480.96 MB / 32.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 15.14.0 - /usr/local/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 7.7.6 - /usr/local/bin/npm
  Browsers:
    Brave Browser: 90.1.23.71
    Firefox: 84.0.1
    Safari: 14.0.3
  npmPackages:
    @sveltejs/kit: next => 1.0.0-next.83
    svelte: ^3.37.0 => 3.37.0
    vite: ^2.1.5 => 2.1.5
```
</details>

**Severity**
Well, it kind of blocks the whole app 🙈

### PR DESCRIPTION
**Pull request by phaleth**
fixes _[edit: unclear that this solves this issue?]_ #1127

also fixes the so called body parsing bug that currently needs to be mitigated by adding the following line to some server side routes
`if (typeof request.body === 'string') request.body = JSON.parse(request.body);`

### Before submitting the PR, please make sure you do the following
- [ ] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [ ] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [ ] Run the tests with `pnpm test` and lint the project with `pnpm lint`

### Changesets
- [ ] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpx changeset` and following the prompts

### CODE REVIEW

**Comment by Rich-Harris** [comment]
Thank you, though I'm unclear on how this would solve #1127? Lowercasing headers for consistency makes sense, but I think it probably needs to happen here... https://github.com/sveltejs/kit/blob/5d8146fd2909ed2828860327a46a226cf7c80f74/packages/kit/src/runtime/server/index.js#L31-L34 ...using this helper: https://github.com/sveltejs/kit/blob/5d8146fd2909ed2828860327a46a226cf7c80f74/packages/kit/src/runtime/server/endpoint.js#L45-L55

Can you add a test that would fail without this PR, and a changeset? The PR template is there for a reason :)

**Comment by phaleth** [comment]
Thank you, @Rich-Harris. Unfortunately, I'm unable to add the test. It's kind of difficult to reproduce the issue. Without this PR I'm getting the same stack trace as posted on the issue #1127, once I visit the site on localhost:3000, but on a project that I cannot disclose.

If I add just `console.log('incoming.headers', incoming.headers);` I get the following.
```
user@debian:~/buggy-site$ npm run dev

> hn-site-mod@0.0.1 dev /home/user/buggy-site
> svelte-kit dev


  SvelteKit v1.0.0-next.85

  local:   http://localhost:3000
  network: not exposed

  Use --host to expose server to other devices on this network


incoming.headers {
  'Content-Type': 'application/json',
  cookie: undefined,
  authorization: undefined
}
TypeError: Cannot read property 'split' of undefined
    at parse_body (file:///home/user/buggy-site/node_modules/@sveltejs/kit/dist/ssr.js:1333:60)
    at ssr (file:///home/user/buggy-site/node_modules/@sveltejs/kit/dist/ssr.js:1456:9)
    at fetch (file:///home/user/buggy-site/node_modules/@sveltejs/kit/dist/ssr.js:700:30)
    at load (/home/user/buggy-site/src/routes/[list]/[page].svelte:13:21)
    at load_node (file:///home/user/buggy-site/node_modules/@sveltejs/kit/dist/ssr.js:788:30)
    at respond (file:///home/user/buggy-site/node_modules/@sveltejs/kit/dist/ssr.js:1010:21)
    at async render_page (file:///home/user/buggy-site/node_modules/@sveltejs/kit/dist/ssr.js:1158:20)
    at async render (file:///home/user/buggy-site/node_modules/@sveltejs/kit/dist/ssr.js:1475:10)
    at async ssr (file:///home/user/buggy-site/node_modules/@sveltejs/kit/dist/ssr.js:1462:10)
    at async Immediate.<anonymous> (file:///home/user/buggy-site/node_modules/@sveltejs/kit/dist/chunks/index.js:3301:23)
```

When I add the same change proposed in this PR and leave the `console.log` in there, I get the following.
```
user@debian:~/buggy-site$ npm run dev

> hn-site-mod@0.0.1 dev /home/user/buggy-site
> svelte-kit dev


  SvelteKit v1.0.0-next.85

  local:   http://localhost:3000
  network: not exposed

  Use --host to expose server to other devices on this network


incoming.headers {
  'content-type': 'application/json',
  cookie: undefined,
  authorization: undefined
}
```

If you need more info, please, let me know.  I realize the PR totally looked like a code butchering attempt at the time of your last comment, but thanks to @rawtaz this PR now looks much better.

**Comment by rawtaz** [comment]
The current changes have been rebased on master and should be conflict-free, but it seems that the [Windows test](https://github.com/sveltejs/kit/pull/1151/checks?check_run_id=2421975502) has a problem launching Chrome: `117: browserType.launch: Failed to launch chromium because executable doesn't exist at C:\Users\runneradmin\AppData\Local\ms-playwright\chromium-857950\chrome-win\chrome.exe`

I'm not sure neither @phaleth nor I can do anything about that, what do you think @Rich-Harris?

File: packages/kit/src/runtime/server/index.js [hunk 1]
```diff
@@ -20,6 +20,10 @@ export async function respond(incoming, options, state = {}) {
 		};
 	}
 
+	if (!incoming.headers['user-agent']) {
```
Comment (benmccann) [inline packages/kit/src/runtime/server/index.js:23]: what's the reason for this check?

Comment (phaleth) [reply packages/kit/src/runtime/server/index.js:23]: Originally, I've put the code for the fix under the following condition...
https://github.com/sveltejs/kit/blob/92d33913e284e00fb20d882253ba5439eeafc407/packages/kit/src/runtime/server/page/load_node.js#L67-L68
...as that's the place where the problem happens. After Rich suggested the `index.js` file, I've put the code in there and added the additional condition, you've pointed at, cause the error happens only with server side requests.

The condition is there to make kind of a differentiation between client side and server side requests and is also there for performance reasons as client side header keys are all lowercase already.

A client could send a request without the 'user-agent' header, so the condition could potentially lead to the similar error in cases where requests without `user-agent` aren't automatically responded with for instance `405` status code upfront.

I'm not fond of modifying the very root file of the project as any addition done there will only make the overall performance of the app worse, but I'll do what you guys suggest in order to fix the problem.

**Comment by Rich-Harris** [comment]
tried to make edits on this branch but it seems github PRs can't properly track branch relationships across forks, and so this turned into a mess of all the recent commits on `master`. So I had to `git merge --squash` locally, which means this _is_ merged, but I still have to close this manually: b4d0d6c6ef4b96f380e7918c35944cda7e6ea1f5

thank you!

**Pull request closed without merging**